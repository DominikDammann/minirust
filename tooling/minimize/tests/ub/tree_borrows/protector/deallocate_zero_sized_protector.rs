//@ compile-flags: --minimize-tree-borrows

// Check that deallocating a zero-sized allocation containing a strongly protected node is UB.

extern crate intrinsics;
use intrinsics::*;

// `x` is strongly protected but covers zero bytes.
fn foo(_x: &mut (), ptr: *mut u8) {
    unsafe { deallocate(ptr, 1, 1) };
}

fn main() {
    let raw = unsafe { allocate(1, 1) } ;
    let x = unsafe { &mut *(raw as *mut ()) };

    foo(x, raw);
}
