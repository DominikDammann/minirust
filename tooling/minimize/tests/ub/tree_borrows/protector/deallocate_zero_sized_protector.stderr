fatal error: UB: Tree Borrows: deallocating strongly protected allocation
